compiler/src/dotty/tools/dotc/transform/LambdaLift.scala

```diff
@@ -338,11 +338,16 @@ object LambdaLift {
             else (topClass, JavaStatic)
           }
           else (lOwner, EmptyFlags)
+        // drop Module because class is no longer a singleton in the lifted context.
+        val nFlags = local.flags &~ Module | Lifted | maybeStatic
+        // keep Public when it is a local class
```

Comment:
Why do local classes need to be public and classes lifted into traits non-final?

Reply:
In the issue #4404, @mbloms said:
> Anonymous classes are compiled without the `public final`.

Therefore, I tried to keep anonymous classes as `public` after lifting. However, I did not consider non-anonymous local classes. I think non-anonymous local classes should not be `public`. I'll correct the code to distinguish anonymous and non-anonymous classes.

At the first, I changed the code simply to keep `final` always. Then, test t4565_1 failed. It was compiled but during the execution, there was an exception:
`Exception in thread "main" java.lang.ClassFormatError: Method A$1 in class T has illegal modifiers: 0x1A`
According to the JVM specification, a method declared inside interface cannot have the final flag, so that I thought that `final` should be dropped when a method is lifted into a trait.